src/gh_search/analyze_urls.py

```diff
@@ -0,0 +1,93 @@
+#!/usr/bin/env python3 -u
+"""Analyze code at URLs to determine reference types."""
+
+
+# Standard Python Libraries
+import re
+import sys
+
+# Third-Party Libraries
+import requests
+
+
+def classify_usage(line):
+    """
+    Classify how tj-actions/changed-files is used in a line.
+
+    Look for a pattern like:
+      uses: tj-actions/changed-files[@<version>]
+    and return the classification:
+      - "unpinned" if no version is provided.
+      - "pinned by sha" if version is a 40-character hexadecimal string.
+      - "pinned by tag" otherwise.
+    """
+    # Regex explanation:
+    #   - Look for "uses:" followed by optional whitespace,
+    #   - then the literal "tj-actions/changed-files"
+    #   - then optionally an "@" and one or more non-whitespace characters.
+    m = re.search(r"uses:\s*tj-actions/changed-files(?:@([^\s]+))?", line)
+    if m:
+        version = m.group(1)
+        if not version:
+            return "unpinned"
+        # Check if version looks like a SHA (40 hex characters)
+        if len(version) == 40 and all(c in "0123456789abcdefABCDEF" for c in version):
```

Comment:
Suggested improvement:
```suggestion
        if re.fullmatch(r"[\da-fA-F]{40}", version):
```